src/modules/catalog.js

```diff
@@ -244,21 +244,27 @@ class Catalog {
     // Handle network timeout if specified
     helpers.applyNetworkTimeout(this.options, networkParameters, controller);
 
-    return fetch(requestUrl, {
-      method: 'PUT',
-      body: JSON.stringify({ items }),
-      headers: {
-        'Content-Type': 'application/json',
-        ...helpers.createAuthHeader(this.options),
-      },
-      signal,
-    }).then((response) => {
+    try {
+      const response = await fetch(requestUrl, {
+        method: 'PUT',
+        body: JSON.stringify({ items }),
+        headers: {
+          'Content-Type': 'application/json',
+          ...helpers.createAuthHeader(this.options),
+        },
+        signal,
+      });
+
+      const body = await response.json();
```

Comment:
Lets move this within the conditional `if (response.ok) {`. If we return an error, it might not be in json so we'll resolve a json-parse error instead of the http error on line 264. Besides, that's the only place we're using `body` anyways.